src/client/mod.rs

```diff
@@ -283,28 +374,70 @@ pub struct WrapData {
+pub enum TokenType {
+    /// Batch tokens are encrypted blobs that carry enough information
+    /// for them to be used for Vault actions, but they require no
+    /// storage on disk to track them.
+    Batch,
+    /// Service tokens are what users will generally think of as
+    /// "normal" Vault tokens.
+    Service,
+    /// Will use the mount's tuned default.
+    Default,
+    /// For a token store this will default to batch, unless the client requests
+    /// a different type at generation time.
+    DefaultBatch,
+    /// For a token store this will default to service, unless the client requests
+    /// a different type at generation time.
+    DefaultService,
+}
+
 /// `AppRole` properties
 #[derive(Deserialize, Debug)]
 pub struct AppRoleProperties {
     /// Require `secret_id` to be presented when logging in using this `AppRole`. Defaults to 'true'.
     pub bind_secret_id: bool,
+    /// The secret IDs generated using this role will be cluster local.
+    pub local_secret_ids: bool,
     /// Comma-separated list of CIDR blocks; if set, specifies blocks of IP addresses which can
     /// perform the login operation.
-    pub bound_cidr_list: String,
-    /// If set, the token generated using this `AppRole` is a periodic token; so long as it is
-    /// renewed it never expires, but the TTL set on the token at each renewal is fixed to the value
-    /// specified here. If this value is modified, the token will pick up the new value at its next
-    /// renewal.
-    pub period: VaultDuration,
-    /// List of policies set on tokens issued via this `AppRole`.
-    pub policies: Vec<String>,
+    pub secret_id_bound_cidrs: Option<Vec<String>>,
     /// Number of times any particular `SecretID` can be used to fetch a token from this `AppRole`,
     /// after which the `SecretID` will expire.
-    pub secret_id_num_uses: u64,
-    /// Duration after which any `SecretID` expires.
+    pub secret_id_num_uses: VaultNumUses,
+    /// Duration in either an integer number of seconds (3600) or an integer time unit (60m) after which any SecretID expires.
     pub secret_id_ttl: VaultDuration,
+    /// List of CIDR blocks; if set, specifies blocks of IP addresses which can authenticate successfully,
+    /// and ties the resulting token to these blocks as well.
+    pub token_bound_cidrs: Vec<String>,
+    /// If set, will encode an explicit max TTL onto the token. This is a hard cap even if token_ttl and
+    /// token_max_ttl would otherwise allow a renewal.
+    pub token_explicit_max_ttl: Option<VaultDuration>,
+    /// If set, the default policy will not be set on generated tokens; otherwise it will be added to
+    /// the policies set in token_policies.
+    pub token_no_default_policy: bool,
     /// Duration after which the issued token can no longer be renewed.
     pub token_max_ttl: VaultDuration,
+    /// The maximum number of times a generated token may be used (within its lifetime); 0 means unlimited.
```

Comment:
This comment needs updating with reference to the new `VaultNumUses` type

Reply:
Good call! Removed mention of 0 means unlimited since `VaultNumUses` explicitly states unlimited now.